lib/src/generate/build_impl.dart

```diff
@@ -120,6 +133,51 @@ class BuildImpl {
     }
   }
 
+  /// Checks if the [_assetGraph] needs to be completely invalidated.
+  ///
+  /// For now this just means looking at the current running program, and seeing
+  /// if any of its sources are newer than the asset graph itself.
+  Future<bool> _shouldInvalidateAssetGraph() async {
+    var shouldRebuild = false;
+    await Future.wait(currentMirrorSystem().libraries.keys.map((Uri uri) async {
```

Comment:
Once `shouldRebuild == true`, we should be able to return `true` without waiting for the results of other futures.

Let's discuss offline.